http.go

```diff
@@ -64,6 +65,71 @@ func (k *Kasse) PostLoginPage(res http.ResponseWriter, req *http.Request) {
 	http.Redirect(res, req, redirect, http.StatusFound)
 }
 
+// GetNewUserPage renders the page to create a new user.
+func (k *Kasse) GetNewUserPage(res http.ResponseWriter, req *http.Request) {
+	res.Header().Set("Content-Type", "text/html")
+
+	if err := ExecuteTemplate(res, TemplateInput{Title: "Create new user", Body: "newUser.html"}); err != nil {
+		k.log.Println("Could not render template:", err)
+		http.Error(res, "Internal error", http.StatusInternalServerError)
+		return
+	}
+}
+
+// PostNewUserPage receives a POST request with username and password and tries
+// to create a new user. It will redirect to the first Flashvalue in the
+// session on success, or to / if none is set and save the authenticated user
+// in the session.
+func (k *Kasse) PostNewUserPage(res http.ResponseWriter, req *http.Request) {
+	username := req.FormValue("username")
+	password := []byte(req.FormValue("password"))
+	confirm := []byte(req.FormValue("confirm"))
+
+	if username == "" || len(password) == 0 || len(confirm) == 0 {
+		// TODO: Write own Error function, that uses a template for better
+		// looking error pages. Also, redirect.
+		http.Error(res, "Neither username nor password can be empty", http.StatusBadRequest)
+		return
+	}
+
+	if !bytes.Equal(password, confirm) {
+		// TODO: Write own Error function, that uses a template for better
+		// looking error pages. Also, redirect.
+		http.Error(res, "Password and confirmation don't match", http.StatusBadRequest)
+	}
+
+	user, err := k.RegisterUser(username, password)
+	if err != nil && err != ErrUserExists {
+		k.log.Printf("Registering user %q failed:%v", username, password, err)
+		// TODO: Write own Error function, that uses a template for better
+		// looking error pages. Also, redirect.
+		http.Error(res, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	if err == ErrUserExists {
+		k.log.Println(err)
+		// TODO: Write own Error function, that uses a template for better
+		// looking error pages. Also, redirect.
+		http.Error(res, "User already exists.", http.StatusUnauthorized)
```

Comment:
I would think that this is a valid request, since new users can't know which users already exist. Therefore, http.StatusForbidden should be used according to (my reading of) the blog post.